### PR TITLE
docs: document webkitdirectory breaking change

### PR DESCRIPTION
**Pull request by erickzhao**
#### Description of Change

Documents the changes made in #18343. Although this was a bug fix to align Electron's behavior with the browser, it's a breaking change for any app that used to use the previous (incorrect) behavior.

cc @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
Approved but lint is failing

**Comment by erickzhao** [comment]
@MarshallOfSound fixed the lint!

**Comment by erickzhao** [comment]
Hope the backport tags here are correct. Should be in the docs for any version >= 7, right?

File: .github/PULL_REQUEST_TEMPLATE/hacktoberfest.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#### Description of Change
```
Comment (jkleinsc) [inline .github/PULL_REQUEST_TEMPLATE/hacktoberfest.md]: I don't think this change should be in this PR.

Comment (erickzhao) [reply .github/PULL_REQUEST_TEMPLATE/hacktoberfest.md]: Oops, that definitely was a mistake.

**Pull request merged**